Services/SciMaterials.Services.API/Services/Files/FileService.cs

```diff
@@ -189,10 +190,11 @@ private async Task<Result<Guid>> WriteToDatabase(FileMetadata metadata, Cancella
         file.ContentTypeId = verifyContentTypeResult.Data.Id;
         file.AuthorId = author.Id;
 
-        // TODO: Many to many error... Разобраться
-        // file.Categories = verifyCategoriesResult.Data ?? new List<Category>();
-        // if (!string.IsNullOrEmpty(metadata.Tags))
-        //     file.Tags = (await GetTagsFromSeparatedStringAsync(_separator, metadata.Tags)).Data;
+        if (verifyCategoriesResult.Data is { })
+            file.Categories = verifyCategoriesResult.Data;
+
+        if (!string.IsNullOrEmpty(metadata.Tags))
+            file.Tags = (await GetTagsFromSeparatedStringAsync(_separator, metadata.Tags)).Data;

```

Comment:
`await`-вызов в скобках лучше сперва поместить в отдельную переменную. Красивее смотреться будет.
```suggestion
        {    
            var tags = await GetTagsFromSeparatedStringAsync(
                _separator, 
                metadata.Tags);
            file.Tags = tags.Data;
        

```